src/Sylius/Component/Core/OrderProcessing/StateResolver.php

```diff
@@ -99,4 +120,15 @@ protected function getShippingState(OrderInterface $order)
 
         return OrderShippingStates::PARTIALLY_SHIPPED;
     }
+
+    /**
+     * @param StateMachineInterface $stateMachine
+     * @param string $transition
+     */
+    private function applyTransition(StateMachineInterface $stateMachine, $transition)
+    {
+        if ($stateMachine->can($transition)) {
+            $stateMachine->apply($transition);
+        }
```

Comment:
The state machine throws an exception by itself, and it's quite detailed. 
My question would be if this method has any real purpose as the check that's done here is also done inside of `apply()` method.